flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java

```diff
@@ -105,7 +105,7 @@ public void testDetachedMode() throws InterruptedException, IOException {
 		}
 
 		//additional sleep for the JM/TM to start and establish connection
-		sleep(2000);
+		sleep(3000);
```

Comment:
Why do we have to increase this timeout?

Reply:
I found that this test can fail sometimes after the fix, therefore adding 1 second to avoid the false positive. It was not a problem before because this test used to pass all the time (because it's not doing the right check).

Reply:
Hmm is this timeout increase not a little bit arbitrary. Can we implement some better waiting strategy? Like waiting for the respective log statement, for example.

Reply:
Hmm, waiting for the log statement might be problematic, because if the job can not start due to other issues, then there will be no log files and the wait can be forever.

Reply:
For that we should have a timeout. If the timeout is triggered, then the test should fail.

A simplest approach would be to periodically check for this condition until you've exceeded the timeout time.

Reply:
Good idea. Address your comment. Please take another look